apps/server/src/modules/files-storage-client/service/copy-files.service.ts

```diff
@@ -1,67 +1,65 @@
 import { Injectable } from '@nestjs/common';
-import { EntityId, IComponentProperties, Lesson, Task } from '@shared/domain';
+import { CopyElementType, CopyHelperService, CopyStatus, CopyStatusEnum, EntityId } from '@shared/domain';
 import { CopyFileDto } from '../dto';
 import { EntityWithEmbeddedFiles } from '../interfaces';
 import { FileParamBuilder, CopyFilesOfParentParamBuilder } from '../mapper';
 import { FilesStorageClientAdapterService } from './files-storage-client.service';
 
 // TODO  missing FileCopyParams  ...passing user instead of userId
 
+export type FileUrlReplacement = {
+	regex: RegExp;
+	replacement: string;
+};
+
 @Injectable()
 export class CopyFilesService {
-	constructor(private readonly filesStorageClientAdapterService: FilesStorageClientAdapterService) {}
+	constructor(
+		private readonly copyHelperService: CopyHelperService,
+		private readonly filesStorageClientAdapterService: FilesStorageClientAdapterService
+	) {}
 
-	// TODO missing Status generation like in the other services
-	async copyFilesOfEntity(
-		originalEntity: EntityWithEmbeddedFiles,
-		copyEntity: EntityWithEmbeddedFiles,
+	async copyFilesOfEntity<T extends EntityWithEmbeddedFiles>(
+		originalEntity: T,
+		copyEntity: T,
 		userId: EntityId
-	): Promise<{ entity: EntityWithEmbeddedFiles; response: CopyFileDto[] }> {
+	): Promise<{
+		fileUrlReplacements: FileUrlReplacement[];
+		fileCopyStatus: CopyStatus;
+	}> {
 		const source = FileParamBuilder.build(originalEntity.getSchoolId(), originalEntity);
 		const target = FileParamBuilder.build(copyEntity.getSchoolId(), copyEntity);
 		const copyFilesOfParentParams = CopyFilesOfParentParamBuilder.build(userId, source, target);
 
-		const response = await this.filesStorageClientAdapterService.copyFilesOfParent(copyFilesOfParentParams);
-		const entity = this.replaceUrlsOfEntity(response, copyEntity);
+		const fileDtos = await this.filesStorageClientAdapterService.copyFilesOfParent(copyFilesOfParentParams);
+		const fileUrlReplacements = this.createFileUrlReplacements(fileDtos);
+		const fileCopyStatus = this.deriveCopyStatus(fileDtos);
 
-		return { entity, response };
+		return { fileUrlReplacements, fileCopyStatus };
 	}
 
-	private replaceUrlsOfEntity(responses: CopyFileDto[], entity: EntityWithEmbeddedFiles): EntityWithEmbeddedFiles {
-		responses.forEach((response) => {
-			if (entity instanceof Lesson) {
-				entity = this.replaceUrlsInLessons(entity, response);
-			} else if (entity instanceof Task) {
-				entity = this.replaceUrlsInTask(entity, response);
-			}
+	private createFileUrlReplacements(fileDtos: CopyFileDto[]): FileUrlReplacement[] {
+		return fileDtos.map((fileDto) => {
```

Comment:
We want to avoid direct returns of values, please put this first to a const with good naming.

Personally i avoid destrcutures for this cases. 
``
const xxxx = fileDtos.map((fileDto) => ({
    regex: new RegExp(`${fileDto.sourceId}.+?"`, 'g'),
    replacement: `${fileDto.id ?? 'fileCouldNotBeCopied'}/${fileDto.name}"`,
}); 
``

Reply:
fixed in BC-2417: https://github.com/hpi-schul-cloud/schulcloud-server/pull/3832/files#diff-3bca06226d82d21babae3b5e8334fe29c1fb56cf5ad103faee3bf47b1022fe11R43